File: backend/tests/vulnerable_samples/sqli/safe_parameterized.py
"""Safe: Properly parameterized query — should NOT be flagged"""
from flask import request
import sqlite3


def search_users():
    name = request.args.get("name", "")
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    # SAFE: parameterized query
    cursor.execute("SELECT * FROM users WHERE name = ?", (name,))
    return cursor.fetchall()

File: backend/tests/vulnerable_samples/sqli/SafeParameterized.java
import java.sql.*;
import javax.servlet.http.HttpServletRequest;

public class SafeParameterized {
    public void handle(HttpServletRequest request, Connection conn) throws Exception {
        String name = request.getParameter("name");
        PreparedStatement ps = conn.prepareStatement("SELECT * FROM users WHERE name = ?");
        ps.setString(1, name);
        ResultSet rs = ps.executeQuery();
    }
}
